ThrowIT/Utility.m

```diff
@@ -6,9 +6,54 @@
 //
 
 #import "Utility.h"
+#import "APIManager.h"
+#import "Party.h"
+#import "TimelineViewController.h"
 
 @implementation Utility
++(void)TakeOrChooseImage:(UIViewController *)viewController withSourceType:(UIImagePickerControllerSourceType)sourceType{
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = viewController;
+    imagePickerVC.allowsEditing = YES;
+    imagePickerVC.sourceType = sourceType;
+    [viewController presentViewController:imagePickerVC animated:YES completion:nil];
+}
+
++(NSMutableArray *)getDistancesFromArray:(NSArray *)array withCompletionHandler:(void (^)(BOOL success ))completion{
+    NSMutableArray *distances = [NSMutableArray new];
+    for(int i = 0; i < array.count; i++)
+    {
+        [distances addObject:PARTYDISTANCELABELPLACEHOLDER];
+    }
+    NSMutableArray *place_Ids = [self initLocationsWithArray:array];
+    for(int i = 0; i < place_Ids.count; i++)
+    {
+        [[APIManager shared] loadDistanceDataIntoArray:distances withDestinationLocation:place_Ids[i] withArrayIndex:i withCount:place_Ids.count withCompletionHandler:^(BOOL success) {
+            if (success)
+            {
+                completion(YES);
+            }
+        }];
+    }
```

Comment:
Do we need two loops for this? Could not we call "[distances addObject:PARTYDISTANCELABELPLACEHOLDER]" before the APIManager call in the same for? Also, i think, the loadDistanceDataIntoArray can receive one object instead of the entire array, and the function can return the final value and in success we can assign the value to the array.